components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java

```diff
@@ -765,19 +763,41 @@ protected void sendResponse(HttpServletRequest request, HttpServletResponse resp
 
         // redirect to the caller
         String redirectURL;
-        String commonauthCallerPath = context.getCallerPath();
-
+        String callerPath = context.getCallerPath();
         try {
-            String queryParamsString = "";
-            if (context.getCallerSessionKey() != null) {
-                queryParamsString = FrameworkConstants.SESSION_DATA_KEY + "=" +
-                        URLEncoder.encode(context.getCallerSessionKey(), "UTF-8");
-            }
+            if (!FrameworkUtils.isAbsoluteURI(callerPath)) {
+                ServiceURLBuilder serviceURLBuilder = ServiceURLBuilder.create().addPath(context.getCallerPath());
+                if (context.getCallerSessionKey() != null) {
+                    serviceURLBuilder.addParameter(FrameworkConstants.SESSION_DATA_KEY, URLEncoder.encode(context
+                            .getCallerSessionKey(), "UTF-8"));
+                }
+
+                if (context.isRequestAuthenticated() && context.isRememberMe()) {
+                    serviceURLBuilder.addParameter("chkRemember", "on");
+                }
 
-            if (StringUtils.isNotEmpty(rememberMeParam)) {
-                queryParamsString += "&" + rememberMeParam;
+                try {
+                    redirectURL = serviceURLBuilder.build().getAbsolutePublicURL();
+                } catch (URLBuilderException e) {
+                    throw FrameworkRuntimeException.error("Error while building redirect URL.", e);
+                }
+            } else {
+                String queryParamsString = "";
+                if (context.getCallerSessionKey() != null) {
+                    queryParamsString = FrameworkConstants.SESSION_DATA_KEY + "=" +
+                            URLEncoder.encode(context.getCallerSessionKey(), "UTF-8");
+                }
+                String rememberMeParam = "";
+
+                if (context.isRequestAuthenticated() && context.isRememberMe()) {
+                    rememberMeParam = rememberMeParam + "chkRemember=on";
```

Comment:
Can't we just assign the value to `rememberMeParam` here as the `rememberMeParam` is an empty string?